# Análisis exploratorio de una base de datos descargada de la plataforma kaggle

In [3]:
import pandas as pd

In [4]:
#lectura de los Datos

df = pd.read_csv('dataR1.csv')
df

Random   Average  of chlorophyll in the plant (ACHP)  \
0         R1                                     34.533468   
1         R1                                     34.489028   
2         R2                                     33.100405   
3         R1                                     34.498319   
4         R3                                     36.297008   
...      ...                                           ...   
29995     R3                                     46.400873   
29996     R3                                     46.395047   
29997     R2                                     32.707725   
29998     R1                                     38.624079   
29999     R3                                     46.409177   

        Plant height rate (PHR)  \
0                     54.566983   
1                     54.567692   
2                     67.067344   
3                     54.559049   
4                     45.588894   
...                         ...   
29995                 50.830830   
29996                 50.838058   
29997                 37.065408   
29998                 37.876323   
29999                 50.841213   

       Average wet weight of the growth vegetative (AWWGV)  \
0                                               1.147449     
1                                               1.149530     
2                                               1.104647     
3                                               1.137759     
4                                               1.363205     
...                                                  ...     
29995                                           0.883813     
29996                                           0.900390     
29997                                           0.909055     
29998                                           1.077557     
29999                                           0.871211     

       Average leaf area of the plant (ALAP)  \
0                                1284.229549   
1                                1284.247744   
2                                1009.208996   
3                                1284.227623   
4                                 981.470310   
...                                      ...   
29995                             744.310093   
29996                             744.307234   
29997                             857.849380   
29998                             948.663424   
29999                             744.298380   

       Average number of plant leaves (ANPL)  Average root diameter (ARD)  \
0                                   4.999713                    16.274918   
1                                   5.024259                    16.269452   
2                                   5.007652                    15.980760   
3                                   4.991501                    16.276710   
4                                   4.003682                    16.979894   
...                                      ...                          ...   
29995                               4.012024                    17.796293   
29996                               4.011764                    17.809462   
29997                               2.992712                    18.200984   
29998                               4.002284                    15.733098   
29999                               4.020517                    17.813723   

        Average dry weight of the root (ADWR)  \
0                                    1.706810   
1                                    1.700930   
2                                    1.185391   
3                                    1.716396   
4                                    0.777428   
...                                       ...   
29995                                1.085930   
29996                                1.108226   
29997                                0.903880   
29998                                1.238818   
29999                                1.098666   

    

In [4]:
type(df)

pandas.core.frame.DataFrame

In [7]:
df.head(7)

Random   Average  of chlorophyll in the plant (ACHP)  \
0     R1                                     34.533468   
1     R1                                     34.489028   
2     R2                                     33.100405   
3     R1                                     34.498319   
4     R3                                     36.297008   
5     R3                                     36.287722   
6     R1                                     34.498869   

    Plant height rate (PHR)  \
0                 54.566983   
1                 54.567692   
2                 67.067344   
3                 54.559049   
4                 45.588894   
5                 45.598585   
6                 54.556247   

   Average wet weight of the growth vegetative (AWWGV)  \
0                                           1.147449     
1                                           1.149530     
2                                           1.104647     
3                                           1.137759     
4                                           1.363205     
5                                           1.353790     
6                                           1.165056     

   Average leaf area of the plant (ALAP)  \
0                            1284.229549   
1                            1284.247744   
2                            1009.208996   
3                            1284.227623   
4                             981.470310   
5                             981.460993   
6                            1284.237645   

   Average number of plant leaves (ANPL)  Average root diameter (ARD)  \
0                               4.999713                    16.274918   
1                               5.024259                    16.269452   
2                               5.007652                    15.980760   
3                               4.991501                    16.276710   
4                               4.003682                    16.979894   
5                               3.998621                    16.991603   
6                               4.992955                    16.267282   

    Average dry weight of the root (ADWR)  \
0                                1.706810   
1                                1.700930   
2                                1.185391   
3                                1.716396   
4                                0.777428   
5                                0.760842   
6                                1.692835   

    Percentage of dry matter for vegetative growth (PDMVG)  \
0                                          18.399982         
1                                          18.398289         
2                                          19.398789         
3                                          18.413613         
4                                          31.423772         
5                                          31.403639         
6                                          18.414688         

   Average root length (ARL)  Average wet weight of the root (AWWR)  \
0                  19.739037                               2.949240   
1                  19.758836                               2.943137   
2                  20.840822                               2.861635   
3                  19.736098                               2.946784   
4                  17.331894                               2.766242   
5                  17.324861                               2.745382   
6                  19.715916                               2.947532   

    Average dry weight of vegetative plants (ADWV)  \
0                                         0.209251   
1                                         0.216154   
2                                         0.200113   
3                                         0.223092   
4                                         0.424172   
5                                         0.435148   
6                                         0.218530   

   Percentage of dry matt

In [6]:
df.tail(5)

Random   Average  of chlorophyll in the plant (ACHP)  \
29995     R3                                     46.400873   
29996     R3                                     46.395047   
29997     R2                                     32.707725   
29998     R1                                     38.624079   
29999     R3                                     46.409177   

        Plant height rate (PHR)  \
29995                 50.830830   
29996                 50.838058   
29997                 37.065408   
29998                 37.876323   
29999                 50.841213   

       Average wet weight of the growth vegetative (AWWGV)  \
29995                                           0.883813     
29996                                           0.900390     
29997                                           0.909055     
29998                                           1.077557     
29999                                           0.871211     

       Average leaf area of the plant (ALAP)  \
29995                             744.310093   
29996                             744.307234   
29997                             857.849380   
29998                             948.663424   
29999                             744.298380   

       Average number of plant leaves (ANPL)  Average root diameter (ARD)  \
29995                               4.012024                    17.796293   
29996                               4.011764                    17.809462   
29997                               2.992712                    18.200984   
29998                               4.002284                    15.733098   
29999                               4.020517                    17.813723   

        Average dry weight of the root (ADWR)  \
29995                                1.085930   
29996                                1.108226   
29997                                0.903880   
29998                                1.238818   
29999                                1.098666   

        Percentage of dry matter for vegetative growth (PDMVG)  \
29995                                          12.812881         
29996                                          12.822240         
29997                                           8.050019         
29998                                          11.953827         
29999                                          12.816439         

       Average root length (ARL)  Average wet weight of the root (AWWR)  \
29995                  18.457811                               3.264946   
29996                  18.457313                               3.259574   
29997                  19.070005                               3.495121   
29998                  19.164418                               3.141850   
29999                  18.455817                               3.270421   

        Average dry weight of vegetative plants (ADWV)  \
29995                                         0.108366   
29996                                         0.129387   
29997                                         0.056698   
29998                                         0.116892   
29999                                         0.116370   

       Percentage of dry matter for root growth (PDMRG) Class  
29995                                         33.343799    TC  
29996                                         33.338906    TC  
29997                                         26.041147    TC  
29998                                         39.125346    TC  
29999                                         33.353872    TC

In [8]:
pd.set_option('display.max_columns',None)

In [10]:
df.head(3)

Random   Average  of chlorophyll in the plant (ACHP)  \
0     R1                                     34.533468   
1     R1                                     34.489028   
2     R2                                     33.100405   

    Plant height rate (PHR)  \
0                 54.566983   
1                 54.567692   
2                 67.067344   

   Average wet weight of the growth vegetative (AWWGV)  \
0                                           1.147449     
1                                           1.149530     
2                                           1.104647     

   Average leaf area of the plant (ALAP)  \
0                            1284.229549   
1                            1284.247744   
2                            1009.208996   

   Average number of plant leaves (ANPL)  Average root diameter (ARD)  \
0                               4.999713                    16.274918   
1                               5.024259                    16.269452   
2                               5.007652                    15.980760   

    Average dry weight of the root (ADWR)  \
0                                1.706810   
1                                1.700930   
2                                1.185391   

    Percentage of dry matter for vegetative growth (PDMVG)  \
0                                          18.399982         
1                                          18.398289         
2                                          19.398789         

   Average root length (ARL)  Average wet weight of the root (AWWR)  \
0                  19.739037                               2.949240   
1                  19.758836                               2.943137   
2                  20.840822                               2.861635   

    Average dry weight of vegetative plants (ADWV)  \
0                                         0.209251   
1                                         0.216154   
2                                         0.200113   

   Percentage of dry matter for root growth (PDMRG) Class  
0                                         57.633906    SA  
1                                         57.633697    SA  
2                                         41.289875    SA

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 14 columns):
 #   Column                                                   Non-Null Count  Dtype  
---  ------                                                   --------------  -----  
 0   Random                                                   30000 non-null  object 
 1    Average  of chlorophyll in the plant (ACHP)             30000 non-null  float64
 2    Plant height rate (PHR)                                 30000 non-null  float64
 3   Average wet weight of the growth vegetative (AWWGV)      30000 non-null  float64
 4   Average leaf area of the plant (ALAP)                    30000 non-null  float64
 5   Average number of plant leaves (ANPL)                    30000 non-null  float64
 6   Average root diameter (ARD)                              30000 non-null  float64
 7    Average dry weight of the root (ADWR)                   30000 non-null  float64
 8    Percentage of dry matter 

In [12]:
df['Random'].unique()

array(['R1', 'R2', 'R3'], dtype=object)

In [14]:
df['Class'].unique()

array(['SA', 'SB', 'SC', 'TA', 'TB', 'TC'], dtype=object)

In [18]:
df['Average wet weight of the growth vegetative (AWWGV)'].unique()

array([1.14744902, 1.1495298 , 1.10464721, ..., 0.90905483, 1.07755726,
       0.87121062])

In [19]:
df.nunique()

Random                                                         3
 Average  of chlorophyll in the plant (ACHP)               30000
 Plant height rate (PHR)                                   30000
Average wet weight of the growth vegetative (AWWGV)        30000
Average leaf area of the plant (ALAP)                      30000
Average number of plant leaves (ANPL)                      30000
Average root diameter (ARD)                                30000
 Average dry weight of the root (ADWR)                     30000
 Percentage of dry matter for vegetative growth (PDMVG)    30000
Average root length (ARL)                                  30000
Average wet weight of the root (AWWR)                      30000
 Average dry weight of vegetative plants (ADWV)            30000
Percentage of dry matter for root growth (PDMRG)           30000
Class                                                          6
dtype: int64

In [20]:
#df.columns.to_list() #me genera una lista de los nombres de las columnas en la tabla

for name in df.columns.to_list():
    print(f'La columna: {name} contiene los elementos {df[name].unique()}')

La columna: Random contiene los elementos ['R1' 'R2' 'R3']
La columna:  Average  of chlorophyll in the plant (ACHP) contiene los elementos [34.53346786 34.48902831 33.10040473 ... 32.70772487 38.6240785
 46.40917724]
La columna:  Plant height rate (PHR) contiene los elementos [54.56698291 54.56769212 67.06734371 ... 37.06540776 37.87632299
 50.84121303]
La columna: Average wet weight of the growth vegetative (AWWGV) contiene los elementos [1.14744902 1.1495298  1.10464721 ... 0.90905483 1.07755726 0.87121062]
La columna: Average leaf area of the plant (ALAP) contiene los elementos [1284.22954908 1284.24774388 1009.20899584 ...  857.84938026  948.66342405
  744.29837955]
La columna: Average number of plant leaves (ANPL) contiene los elementos [4.99971308 5.02425915 5.00765198 ... 2.99271194 4.00228364 4.02051656]
La columna: Average root diameter (ARD) contiene los elementos [16.27491791 16.26945189 15.98075994 ... 18.20098421 15.733098
 17.81372278]
La columna:  Average dry weight of t

In [21]:
#Eliminar columnas que no brindan informacion
valores_unicos_n = df.nunique()

columnas_sn_info = valores_unicos_n[valores_unicos_n == 1].index.to_list()
columnas_sn_info

[]

In [22]:
df_clean = df.drop(columnas_sn_info,axis=1)

In [55]:
#df_clean = df_clean.drop('JobSatisfaction', axis=1) #ejemplo de eliminacion de una columna en especifico

In [23]:
df.head(3)

Random   Average  of chlorophyll in the plant (ACHP)  \
0     R1                                     34.533468   
1     R1                                     34.489028   
2     R2                                     33.100405   

    Plant height rate (PHR)  \
0                 54.566983   
1                 54.567692   
2                 67.067344   

   Average wet weight of the growth vegetative (AWWGV)  \
0                                           1.147449     
1                                           1.149530     
2                                           1.104647     

   Average leaf area of the plant (ALAP)  \
0                            1284.229549   
1                            1284.247744   
2                            1009.208996   

   Average number of plant leaves (ANPL)  Average root diameter (ARD)  \
0                               4.999713                    16.274918   
1                               5.024259                    16.269452   
2                               5.007652                    15.980760   

    Average dry weight of the root (ADWR)  \
0                                1.706810   
1                                1.700930   
2                                1.185391   

    Percentage of dry matter for vegetative growth (PDMVG)  \
0                                          18.399982         
1                                          18.398289         
2                                          19.398789         

   Average root length (ARL)  Average wet weight of the root (AWWR)  \
0                  19.739037                               2.949240   
1                  19.758836                               2.943137   
2                  20.840822                               2.861635   

    Average dry weight of vegetative plants (ADWV)  \
0                                         0.209251   
1                                         0.216154   
2                                         0.200113   

   Percentage of dry matter for root growth (PDMRG) Class  
0                                         57.633906    SA  
1                                         57.633697    SA  
2                                         41.289875    SA

In [24]:
df_clean.describe()

Average  of chlorophyll in the plant (ACHP)   Plant height rate (PHR)  \
count                                  30000.000000              30000.000000   
mean                                      38.060575                 59.020331   
std                                        4.441682                 11.985213   
min                                       32.664004                 37.020398   
25%                                       34.399604                 51.059436   
50%                                       36.575753                 55.912058   
75%                                       42.300836                 67.069465   
max                                       46.431668                 77.040610   

       Average wet weight of the growth vegetative (AWWGV)  \
count                                       30000.000000     
mean                                            1.247257     
std                                             0.245989     
min                                             0.848351     
25%                                             1.064865     
50%                                             1.211328     
75%                                             1.501692     
max                                             1.775383     

       Average leaf area of the plant (ALAP)  \
count                           30000.000000   
mean                             1063.639833   
std                               260.712823   
min                               658.488225   
25%                               875.501356   
50%                              1009.249081   
75%                              1243.866525   
max                              1751.031654   

       Average number of plant leaves (ANPL)  Average root diameter (ARD)  \
count                           30000.000000                 30000.000000   
mean                                3.944789                    16.285792   
std                                 0.702537                     2.882793   
min                                 2.959173                    11.075737   
25%                                 3.013099                    13.933350   
50%                                 3.998577                    16.248496   
75%                                 4.015829                    18.202577   
max                                 5.037358                    23.325870   

        Average dry weight of the root (ADWR)  \
count                            30000.000000   
mean                                 1.000981   
std                                  0.456292   
min                                  0.241972   
25%                                  0.675892   
50%                                  0.833799   
75%                                  1.313504   
max                                  2.196556   

        Percentage of dry matter for vegetative growth (PDMVG)  \
count                                       30000.000000         
mean                                           22.674553         
std                                             9.571091         
min                                             8.021505         
25%                                            14.980983         
50%                                            21.954006         
75%                                            27.711471         
max                                            43.660373         

       Average root length (ARL)  Average wet weight of the root (AWWR)  \
count               30000.000000                           30000.000000   
mean                   18.071061                               2.907011   
std                     2.932830                               1.361928   
min                    12.359924                               1.126715   
25%                    15.600294                               1.911408   
50%                    18.477019                               2.818611   
75%                    20.3652

Plantas con promedio de clorofila > 40

In [33]:
clorofila_promedio =df_clean[(df_clean[' Average  of chlorophyll in the plant (ACHP)'] >= 46)]

In [34]:
clorofila_promedio.describe()

Average  of chlorophyll in the plant (ACHP)   Plant height rate (PHR)  \
count                                   1684.000000               1684.000000   
mean                                      46.399881                 50.840212   
std                                        0.010097                  0.010334   
min                                       46.361027                 50.798257   
25%                                       46.392865                 50.833168   
50%                                       46.399779                 50.840266   
75%                                       46.406906                 50.847152   
max                                       46.431668                 50.869418   

       Average wet weight of the growth vegetative (AWWGV)  \
count                                        1684.000000     
mean                                            0.880867     
std                                             0.009828     
min                                             0.848351     
25%                                             0.874414     
50%                                             0.881213     
75%                                             0.887393     
max                                             0.920243     

       Average leaf area of the plant (ALAP)  \
count                            1684.000000   
mean                              744.301739   
std                                 0.009949   
min                               744.266798   
25%                               744.294949   
50%                               744.301669   
75%                               744.308502   
max                               744.332226   

       Average number of plant leaves (ANPL)  Average root diameter (ARD)  \
count                            1684.000000                  1684.000000   
mean                                4.000132                    17.810134   
std                                 0.010299                     0.010344   
min                                 3.963553                    17.777433   
25%                                 3.993440                    17.803226   
50%                                 3.999959                    17.810091   
75%                                 4.007008                    17.817306   
max                                 4.036282                    17.843241   

        Average dry weight of the root (ADWR)  \
count                             1684.000000   
mean                                 1.086198   
std                                  0.009810   
min                                  1.055029   
25%                                  1.079441   
50%                                  1.085933   
75%                                  1.093105   
max                                  1.117733   

        Percentage of dry matter for vegetative growth (PDMVG)  \
count                                        1684.000000         
mean                                           12.825987         
std                                             0.009928         
min                                            12.789803         
25%                                            12.819169         
50%                                            12.826242         
75%                                            12.832681         
max                                            12.853244         

       Average root length (ARL)  Average wet weight of the root (AWWR)  \
count                1684.000000                            1684.000000   
mean                   18.459556                               3.255095   
std                     0.010044                               0.009649   
min                    18.422190                               3.220282   
25%                    18.452802                               3.248434   
50%                    18.459503                               3.255124   
75%                    18.4663

In [61]:
Relacion_clorofila_foliar_raiz = df_clean[(df_clean[' Average  of chlorophyll in the plant (ACHP)'] >= 20) &
                                          (df_clean['Average leaf area of the plant (ALAP)'] >= 1000) &
                                          (df_clean['Average root length (ARL)'] >= 20)]

In [62]:
Relacion_clorofila_foliar_raiz.describe()

Average  of chlorophyll in the plant (ACHP)   Plant height rate (PHR)  \
count                                   8285.000000               8285.000000   
mean                                      34.771541                 72.115742   
std                                        1.205632                  4.407836   
min                                       33.066194                 66.412321   
25%                                       34.193178                 67.063920   
50%                                       34.398843                 74.900035   
75%                                       35.606619                 75.017194   
max                                       36.631543                 77.040610   

       Average wet weight of the growth vegetative (AWWGV)  \
count                                        8285.000000     
mean                                            1.300033     
std                                             0.175260     
min                                             1.074360     
25%                                             1.132606     
50%                                             1.239392     
75%                                             1.505864     
max                                             1.543718     

       Average leaf area of the plant (ALAP)  \
count                            8285.000000   
mean                             1339.215850   
std                               246.277258   
min                              1009.190277   
25%                              1243.860425   
50%                              1253.960242   
75%                              1435.626947   
max                              1751.031654   

       Average number of plant leaves (ANPL)  Average root diameter (ARD)  \
count                            8285.000000                  8285.000000   
mean                                4.396356                    17.816668   
std                                 0.489257                     3.282041   
min                                 3.963477                    13.904618   
25%                                 3.997795                    15.956841   
50%                                 4.009275                    16.299576   
75%                                 4.996371                    19.735737   
max                                 5.035298                    23.325870   

        Average dry weight of the root (ADWR)  \
count                             8285.000000   
mean                                 1.236842   
std                                  0.531694   
min                                  0.638796   
25%                                  0.768885   
50%                                  1.174073   
75%                                  1.430104   
max                                  2.196556   

        Percentage of dry matter for vegetative growth (PDMVG)  \
count                                        8285.000000         
mean                                           27.005917         
std                                             9.258222         
min                                            17.615324         
25%                                            19.395332         
50%                                            26.602474         
75%                                            27.099023         
max                                            43.660373         

       Average root length (ARL)  Average wet weight of the root (AWWR)  \
count                8285.000000                            8285.000000   
mean                   21.478731                               3.823428   
std                     1.165087                               1.764394   
min                    20.338130                               1.906142   
25%                    20.475568                               2.793070   
50%                    20.838983                               2.844863   
75%                    22.5561

conclusion:  la media del promedio de clorofila que presenta una planta cuya clorofila es mayor a 20, es de 34.77
            la longitud promedio de la raiz es de 21.47 de las 8285 plantas cuya longitud de raiz es mayor a 20
            el area foliar de las plantas mayores a 1000 en promedio es de 1339.21